clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -578,10 +580,12 @@ private ControllerClient createLivenessHeartbeatControllerClient(VenicePropertie
         isSslEnabled(),
         properties.getString(SSL_FACTORY_CLASS_NAME, DEFAULT_SSL_FACTORY_CLASS_NAME),
         this.sslProperties);
-    String veniceControllerUrl = props.getString(VENICE_DISCOVER_URL_PROP);
+    String veniceControllerUrlForLivenessHeartbeat = props.getString(VENICE_CHILD_CONTROLLER_PROP);
     String heartbeatStoreClusterName = discoverCluster(heartbeatStoreName);
-    ControllerClient heartbeatStoreControllerClient =
-        ControllerClient.constructClusterControllerClient(heartbeatStoreClusterName, veniceControllerUrl, sslFactory);
+    ControllerClient heartbeatStoreControllerClient = ControllerClient.constructClusterControllerClient(
```

Comment:
Hmm sorry I don't have much context for this - is this creating a controller client to talk to child fabric?

Reply:
Yes, previously VPJ only has parent controller url, so this heartbeat store can only talk to parent controller to request topic to write, but parent controller has a safeguard preventing stream topic requesting by throwing exception like` Parent request topic is disabled!!`. So we can not enable this store with native replication. Instead, we plan to enable AA for it. So we need to add child controller url to VPJ.  In prod, we might use lva1 as WAR is in lva1.

Reply:
Got it..thanks for the explanation!